pkgs/test_core/lib/src/runner/configuration/args.dart

```diff
@@ -177,41 +177,35 @@ Configuration parse(List<String> args) => _Parser(args).parse();
 
 void _parseTestSelection(
     String option, Map<String, Set<TestSelection>> selections) {
-  var firstQuestion = option.indexOf('?');
-  TestSelection selection;
-  String path;
-  if (firstQuestion == -1) {
-    path = option;
-    selection = TestSelection();
-  } else if (option.substring(0, firstQuestion).contains('\\')) {
-    throw FormatException(
-        'When passing test path queries, you must pass the path in URI '
-        'format (use `/` for directory separators instead of `\\`).');
-  } else {
-    final uri = Uri.parse(option);
+  final uri = Uri.parse(option);
+  // Restore the path to how it was given (namely, we want to report paths
```

Comment:
This comment confuses me.

I don't see how it corresponds to "how it was given", since we lose any knowledge about the original format once it is a URI.

I also don't know what you mean by "original separators". I don't see anything that changes separators below. I would expect `File.fromUri(uri).path` to change to platform separators, but I wouldn't expect `decodeComponent` to make any change to separators.

Reply:
FWIW I suspect `File.fromUri(uri).path` might also handle the drive letter thing?

Reply:
Fort the record I don't necessarily think that using the platform separator is an improvement - the changes in the test files look fine to me either way. I'm mostly confused about how the comment lines up with the implementation.

Reply:
> I don't see how it corresponds to "how it was given", since we lose any knowledge about the original format once it is a URI.

In the case of the test that I added, when that uri is parsed it encodes the backslashes and converts them to %5C. So decoding the component restores those encoded backslashes to their original form.

> I also don't know what you mean by "original separators". I don't see anything that changes separators below. I would expect `File.fromUri(uri).path` to change to platform separators, but I wouldn't expect `decodeComponent` to make any change to separators.

Yes I am not explicitly dealing with separators, just ensuring we decode them if they were encoded.

Reply:
> FWIW I suspect `File.fromUri(uri).path` might also handle the drive letter thing?

It doesn't appear to, and it also doesn't support fragments or query strings

Reply:
Note that calling `toFilePath()` would deal with the drive letter issue. But that also fails if there is a fragment or query string, and it would always return a platform specific path, which I think we don't really want.

Reply:
The encoded backslashes is only happening on github actions, but I think its reason enough to leave the decoding logic in either way, so I restored it.